extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -136,6 +152,16 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
   /** @override */
   extractCreativeAndSignature(responseText, responseHeaders) {
     setGoogleLifecycleVarsFromHeaders(responseHeaders, this.lifecycleReporter_);
+    if (responseHeaders.has(AMP_ANALYTICS_HEADER)) {
```

Comment:
Can this be factored into shared code between AdSense and Doubleclick impls, similar to `setGoogleLifecycleVarsFromHeaders`?  Maybe in ads/google/a4a/utils.js?  Ditto below for `onCreativeRender`.